community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/cache/idmapping/string/ParallelSort.java

```diff
@@ -166,11 +164,11 @@ public ParallelSort( Radix radix, LongArray dataCache, long highestSetIndex,
         return rangeParams;
     }
 
-    private String dumpBuckets( int[][] rangeParams, int[] bucketRange, int[] bucketIndex )
+    private String dumpBuckets( long[][] rangeParams, int[] bucketRange, long[] bucketIndex )
```

Comment:
Yes, the array sizes will be bound by number of configured threads, which will be number of available processors, which should be small at least a couple of years into the future.